UhooiPicBook/SceneDelegate.swift

```diff
@@ -55,3 +61,34 @@ class SceneDelegate: UIResponder, UIWindowSceneDelegate {
     }
 
 }
+
+// MARK: - UserActivity
+extension SceneDelegate {
+
+    func scene(_ scene: UIScene, continue userActivity: NSUserActivity) {
+        self.executeUserActivity(userActivity)
+    }
+
+    private func executeUserActivity(_ userActivity: NSUserActivity) {
+        switch userActivity.activityType {
+        case CSSearchableItemActionType:
+            self.executeSpotlightActivity(userActivity)
+        default:
+            return
+        }
+    }
+
+    private func executeSpotlightActivity(_ userActivity: NSUserActivity) {
+        let decoder = JSONDecoder()
+        decoder.keyDecodingStrategy = .convertFromSnakeCase
+
+        guard let userDefaultsKey = userActivity.userInfo?[CSSearchableItemActivityIdentifier] as? String,
+            let data = UserDefaults.standard.value(forKey: userDefaultsKey) as? Data,
+            let monster = try? decoder.decode(MonsterEntity.self, from: data) else {
+                return
+        }
+
+        let vc = MonsterDetailRouter.assembleModule(monster: monster)
+        (self.window?.rootViewController as? UINavigationController)?.pushViewController(vc, animated: true)
```

Comment:
この処理だとすでにモンスター詳細画面を表示しているときにスポットライトから表示すると、二重に表示されてしまうので、ナビゲーションコントローラーの一番上がモンスター詳細画面だったら剥がす処理が必要です！

```suggestion
        guard let nav = self.window?.rootViewController as? UINavigationController else {
            return
        }
        if (nav.viewControllers.last as? MonsterDetailViewController) != nil {
            nav.viewControllers.removeLast()
        }
        let vc = MonsterDetailRouter.assembleModule(monster: monster)
        nav.pushViewController(vc, animated: true)
```

Reply:
もっといい剥がし方があればご提案お願いしますm(_ _)m笑

Reply:
これに関しては剥がすかどうか結構迷いましたw
自分もその仕様だとそんな感じにすると思います！

Reply:
なるほど、気づいた上でこの仕様にしていたのですねｗ
剥がさないと違和感があるので、これでCommitしていただけてOKでした！

Reply:
強いて言うのであれば、
```swift
if (nav.viewControllers.last as? MonsterDetailViewController) != nil {
    nav.viewControllers.removeLast()
}
```
このif文判定ですがクラス判定をするのであれば、以下のやり方の方が適切だと思ったので修正しておきました！
```swift
if nav.viewControllers.last is MonsterDetailViewController {
    nav.viewControllers.removeLast()
}
```